src/commonMain/kotlin/fr/acinq/lightning/message/OnionMessages.kt

```diff
@@ -52,24 +52,32 @@ object OnionMessages {
         }
     }
 
+    fun buildRouteToRecipient(
```

Comment:
Why don't you simply change `buildRoute` below to return a `RouteBlinding.BlindedRouteDetails`? It's a bit confusing to introduce yet another function that does just a subset of it?

Reply:
That's because in case destination is another blinded route we don't have the information we need, we can only have it when we build the last payloads ourselves.

Reply:
But that's not visible in our codepaths though? On the caller side, we indeed only derive the private key when we create an offer with a `Recipient`, so it doesn't matter whether we call `buildRoute` or `buildRouteToRecipient`. The `BlindedRouteDetails` from `buildRoute` are still valid blinded route details, it's just that you shouldn't derive a key from it for the same use-cases if it's a blinded path?

Reply:
I don't understand what's the `BlindedRouteDetails` from `buildRoute`, if `destination: Destination.BlindedPath`, we can't reconstruct the `BlindedRouteDetails`.

Reply:
I don't understand what you mean, `BlindedRouteDetails` are simply a blinded route with the blinding key also exposed. But it's not important, let's go with this and we can refactor later.

Reply:
What I'm saying is that if we didn't build the full route ourselves, we will know the blinded route without knowing the blinding key for the last node. And `buildRoute` handles such cases so it can't return the blinding key for the last node in the general case. However in the subcase where we know the node id of the recipient and build the full route ourselves, then we know the blinding key of the last node and can return it, that's what I do in `buildRouteToRecipient`.